doordeck-sdk/src/mingwMain/resources/doordeck_headless_sdk/DoordeckSdk.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System.Runtime.InteropServices;
+using Doordeck.Headless.Sdk.Model;
+using Doordeck.Headless.Sdk.Wrapper;
+
+namespace Doordeck.Headless.Sdk;
+
+public unsafe class DoordeckSdk(ApiEnvironment apiEnvironment, string? authToken)
+{
+    private readonly Doordeck_Headless_Sdk_ExportedSymbols* _symbols = Methods.Doordeck_Headless_Sdk_symbols();
+
+    private Doordeck_Headless_Sdk_kref_com_doordeck_multiplatform_sdk_api_model_ApiEnvironment _apiEnvironment;
+    private Doordeck_Headless_Sdk_kref_com_doordeck_multiplatform_sdk_KDoordeckFactory _factory;
+    private Doordeck_Headless_Sdk_kref_com_doordeck_multiplatform_sdk_Doordeck _sdk;
+
+    private readonly Account _account = new();
+    private readonly Accountless _accountless = new();
+    private readonly Fusion _fusion = new();
+    private readonly Helper _helper = new();
+    private readonly LockOperations _lockOperations = new();
+    private readonly Platform _platform = new();
+    private readonly Sites _sites = new();
+    private readonly Tiles _tiles = new();
+    private readonly ContextManager _contextManager = new();
+    private readonly CryptoManager _cryptoManager = new();
+
+    public void Initialize()
+    {
+        var token = authToken != null ? Utils.Utils.ToSByte(authToken) : null;
+        _apiEnvironment = apiEnvironment switch
+        {
+            ApiEnvironment.DEV => _symbols->kotlin.root.com.doordeck.multiplatform.sdk.api.model.ApiEnvironment.DEV
+                .get(),
+            ApiEnvironment.STAGING => _symbols->kotlin.root.com.doordeck.multiplatform.sdk.api.model.ApiEnvironment
+                .STAGING.get(),
+            ApiEnvironment.PROD => _symbols->kotlin.root.com.doordeck.multiplatform.sdk.api.model.ApiEnvironment.PROD
+                .get(),
+            _ => _apiEnvironment
+        };
+
+        _factory = _symbols->kotlin.root.com.doordeck.multiplatform.sdk.KDoordeckFactory._instance();
+
+        if (authToken != null)
+        {
+            _sdk = _symbols->kotlin.root.com.doordeck.multiplatform.sdk.KDoordeckFactory.initializeWithAuthToken(_factory, _apiEnvironment, token);
+            Marshal.FreeHGlobal((IntPtr)token);
```

Comment:
If `initializeWithAuthToken` throws an exception, will that cause a memory leak on `token` because the new line doesn't get evaluated? Most other usages of this seem to be in a finally block.

Reply:
Yup, good catch!